traffic_generator_trex.py

```diff
@@ -377,6 +377,80 @@ def configure_traffic_stream(self, traffic_flows, nr_of_flows,
             self.__traffic_flows = traffic_flows
             return True
         elif traffic_flows == TrafficFlowType.none:
+            self.__traffic_flows = traffic_flows
+            return True
+        elif traffic_flows == TrafficFlowType.vxlan_l3_ipv4:
+            tunnel_src_mac = "00:00:00:00:00:01"
+            tunnel_dst_mac = kwargs.pop("tunnel_dst_mac", None)
+
+            L2 = Ether(src=tunnel_src_mac,
+                       dst=tunnel_dst_mac)
+            L3 = IP(src="3.1.1.2", dst="3.1.1.1")
+            L4 = UDP(sport=32768, dport=4789, chksum=0)
+            L5 = VXLAN(vni=69, NextProtocol=0, flags='Instance')
+            L6 = Ether(src=trex_src_mac, dst=trex_dst_mac)
+            L7 = IP(src="1.0.0.0", dst="2.0.0.0")
+            tmpl = L2/L3/L4/L5/L6/L7
+
+
+            if (len(tmpl) + 4) > packet_size:  # +4 for Ethernet CRC
+                raise ValueError("Packet size ({} bytes) too small for"
+                                 "requested packet ({} bytes)!".
+                                 format(packet_size, len(tmpl) + 4))
+
+            src_end = str(netaddr.IPAddress(
+                int(netaddr.IPAddress('1.0.0.0')) +
+                nr_of_flows - 1))
+            dst_end = str(netaddr.IPAddress(
+                int(netaddr.IPAddress('2.0.0.0')) +
+                nr_of_flows - 1))
+            port_start = 32768
+            port_end = 32800
```

Comment:
From the RFC the port numbers should be in the 49152-65535 range, but should also flow specific. So I guess below you should assign the same UDP port to each unique pair of source/dest IP addresses.